vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -349,14 +351,7 @@ public int readBytesAvailable() {
     public ByteBuf readBlocking() throws IOException {
         synchronized (request.connection()) {
             while (input1 == null && !eof) {
-                try {
```

Comment:
This is broken, there is different reading and writing versions with different state variabled (waitingForRead and waitingForWrite)

Reply:
Oops, my IDE proposed the change. I am reverting this change, it is not in the scope of the PR also. Thanks.

Reply:
Reverting done. Thanks for spotting this @stuartwdouglas , I should be more careful next time :-)